### PR TITLE
Fixing broken exception handling after GAX 0.13.0 upgrade.

### PR DESCRIPTION
**Pull request by dhermes**
~~This PR is currently incomplete (hence the "don't merge").~~

I wanted @bjwatson @geigerj et al to see what fix was needed.

This may be broken from https://github.com/googleapis/gax-python/pull/127 or this may be the intended behavior.

I filed https://github.com/googleapis/gax-python/issues/129 and https://github.com/googleapis/gax-python/issues/130 as well

### CODE REVIEW

File: google/cloud/logging/_gax.py [hunk 1]
```diff
@@ -18,18 +18,17 @@
...
 from google.gax import CallOptions
 from google.gax import INITIAL_PAGE
-from google.gax.errors import GaxError
 from google.logging.type.log_severity_pb2 import LogSeverity
 from google.logging.v2.logging_config_pb2 import LogSink
 from google.logging.v2.logging_metrics_pb2 import LogMetric
 from google.logging.v2.log_entry_pb2 import LogEntry
 from google.protobuf.json_format import Parse
 from grpc import StatusCode
+from grpc._channel import _Rendezvous
```
Comment (tseaver) [inline google/cloud/logging/_gax.py]: Argggh!  Are we really going to commit to scattering use of that private name everywhere?  Could we mitigate by importing it into `google.cloud.exceptions`, and then aliasing it with a name like `GrpcRendezvous`?

File: google/cloud/pubsub/_gax.py [hunk 2]
```diff
@@ -520,7 +519,7 @@ def make_gax_publisher_api(connection):
     """
     channel = None
     if connection.in_emulator:
-        channel = insecure_channel(connection.host, None)
+        channel = insecure_channel(connection.host)
```
Comment (tseaver) [inline google/cloud/pubsub/_gax.py]: What happened here?  OIC, `None` is the default -- did that change?

Comment (dhermes) [inline google/cloud/pubsub/_gax.py]: [`beta`][1] vs. [stable][2]

[1]: https://github.com/grpc/grpc/blob/ee43d7bbb11a36c91c6f1ebff6bf3da70b7546b2/src/python/grpcio/grpc/beta/implementations.py#L105
[2]: https://github.com/grpc/grpc/blob/ee43d7bbb11a36c91c6f1ebff6bf3da70b7546b2/src/python/grpcio/grpc/__init__.py#L1180

File: google/cloud/logging/_gax.py (hunk 1 above)
Comment (dhermes) [inline google/cloud/logging/_gax.py]: > Are we really going to commit

Was hoping to resolve discussion in #2156. Happy to add an alias of our own.

I think the real issue though is 0.13.0 of GAX. Like I think they meant to be catching those `_Rendezvous` and giving us `GaxError`s to catch.

**Comment by tseaver** [comment]
@dhermes do we need to push this through to get green, and then work on resolving the issue with `GaxError` / `_Rendezvous`?

**Comment by dhermes** [comment]
That's fine with me. Definitely need to make sure @bjwatson and @geigerj are aware of what has broken

**Comment by bjwatson** [comment]
Thanks @dhermes and @tseaver. @geigerj and I will look into https://github.com/googleapis/gax-python/issues/130 on Monday. IIUC, this does not block the 0.19.0 release, correct?

**Comment by dhermes** [comment]
@bjwatson It won't block us from releasing, but it's likely something you'll want to deal with ASAP (but in parallel, not in serial).

**Comment by tseaver** [comment]
@dhermes Let's go ahead and merge, and let #2156 be where we track the issue about using `_Rendezvous`.

**Comment by bjwatson** [comment]
@dhermes Agreed. Thanks.

**Comment by dhermes** [comment]
@tseaver PTAL. I added a standalone commit for the rename. Let's get this merged ASAP.

I also realized `_Rendezvous` is used as a non-exception when `stream` type responses come back from gRPC.

**Comment by dhermes** [comment]
Hoping to go green finally:
https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/161085138

**Comment by daspecster** [comment]
@dhermes some failures :(
https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/161085138#L1238-L1283

**Comment by daspecster** [comment]
It's almost like Logging got turned off or is having an outage?

**Comment by dhermes** [comment]
Do they work locally for you?

**Comment by daspecster** [comment]
I haven't tried but I'll do that right now.

**Comment by daspecster** [comment]
Nope doesn't work for me either.
I ran `$ tox -e system-tests -- logging`

**Comment by dhermes** [comment]
Yeah I think that's the problem. @tseaver Can you check out https://console.cloud.google.com/apis/credentials?project=precise-truck-742 and make sure everything is in order permissions wise? (We recently have a new JSON keyfile on Travis since the old one didn't have the project ID in the JSON. #2292 added this change)

**Comment by daspecster** [comment]
I got mine working, it was the permissions.

With regards to the console link you sent...I don't have permissions for that project.
screen shot 2016-09-19 at 1 18 55 pm

The place to update it would be https://console.cloud.google.com/iam-admin/iam/project?project=precise-truck-742

**Comment by dhermes** [comment]
> I don't have permissions for that project.

Wasn't sure if you did, hence: "@tseaver Can you check out"

**Comment by daspecster** [comment]
Sorry, thought I'd give it a go since it took me some hunting and pecking in the UI.

**Comment by tseaver** [comment]
The perms look right to me:

screenshot from 2016-09-19 13 28 18

**Comment by dhermes** [comment]
> Sorry, thought I'd give it a go since it took me some hunting and pecking in the UI.

No worries. Thanks for trying.

**Comment by dhermes** [comment]
I just used that exact key locally and had no issues:

```
TRAVIS_BUILD_ID=161085138 tox -e system-tests -- logging
```

I restarted https://travis-ci.org/GoogleCloudPlatform/google-cloud-python/builds/161085138, hopefully it will go green

**Comment by dhermes** [comment]
`tox -e system-tests` has passed already on the rebuild, hopefully `system-tests3` will follow. w00t

**Comment by dhermes** [comment]
w00t. Green!

**Pull request merged**